### PR TITLE
Ci/workflow

### PR DESCRIPTION
**Pull request by geolffreym**
* ci: static analysis
* ci: test

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install
+      - name: Test
+        run: |
+          make test-coverage
```
[the remark this answers is not in the document]
Comment (aphelionz) [reply .github/workflows/test.yml:28]: If we're going to do test coverage we should output the HTML report as a build artifact so we can review it via the GH UI

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	$(FLAKE8) $(PYTHON_MODULES)
+test: check-coding-style
+	$(PYTEST) $(PYTHON_MODULES)
+test-coverage: check-coding-style test
+	$(COVERAGE) run --source=./src $(VENV)/bin/py.test
+	$(COVERAGE) report
+test-check:
+	$(PYTEST) $(PYTHON_MODULES)
+
+.PHONY: default venv requirements bootstrap check-coding-style test test-coverage
```
Comment (aphelionz) [inline Makefile:43]: Needs a newline

File: README.md [hunk 3]
```diff
@@ -51,13 +51,33 @@ class Dummy:
...
 
-See [SCHEME.md](https://github.com/ZorrillosDev/watchit-gateway/blob/master/SCHEME.md).
+* [SCHEME.md](https://github.com/ZorrillosDev/watchit-gateway/blob/master/SCHEME.md).
 
-## Usage
+* [USAGE.md](https://github.com/ZorrillosDev/watchit-gateway/blob/master/USAGE.md).
+
+
+## Development
+ Please make sure you have `make` installed
```
Comment (aphelionz) [inline README.md:62]: See my comment on GNU vs BSD Make above

File: docker-compose.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 version: '3'
 services:
-  watchit_ipfs:
+  watchit-ipfs:
     image: ipfs/go-ipfs:v0.8.0
```
Comment (aphelionz) [inline docker-compose.yml:4]: Commenting to make a future issue: Upgrade to 0.9.0

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
```
Comment (aphelionz) [inline pyproject.toml:3]: Newline

File: src/core/helper/tests/test_cache.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+def test_strReplace():
+    string = "Hello, World!"
+    assert string.replace("H", "J") == "Jello, World!"
```
Comment (aphelionz) [inline src/core/helper/tests/test_cache.py:3]: Can this be removed?

**Review by aphelionz — CHANGES_REQUESTED** [review]
Overall looks great, I would configure your editor to make sure each file has a newline at the end according to the POSIX standard and re-save.

File: src/core/helper/tests/test_cache.py (hunk 6 above)
Comment (geolffreym) [reply src/core/helper/tests/test_cache.py:3]: sure, i all add some real tests there XD

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,43 @@
+PYTHON_MODULES := src
```
[the remark this answers is not in the document]
Comment (geolffreym) [reply Makefile]: Make file based on common syntax!!

**Pull request merged**